### ISSUE TITLE
Recognize and run code blocks

### ISSUE THREAD

**Issue by ngharrison** [apommel/vscode-matlab-interactive-terminal#13]
Just a feature suggestion. The ability to run blocks of code delimited by `%%` would be very useful. Thanks for the great extension!

**Comment by apommel** [comment]
Thanks for your feedback! You mean that this feature would be able to run blocks one after the other, a bit like Jupyter?

**Comment by ngharrison** [comment]
Yes, similar to that. The idea of code blocks or code cells. The Matlab IDE does this itself using `%%`.
There is similar functionality in Python IDEs (Jupyter, Spyder, VSCode Python extension) with some delimiter such as `#%%`.

One more thing I'd add on is to have the focus stay in the editor when code (selection, line, block, etc.) is run. Currently it always takes me to the terminal and I have to click back into the editor to run the next line.

These would be nice features. I'm not sure how complicated they would be.

**Comment by aidedinertial** [comment]
Here is my workaround for cell evaluation:

This solution requires the Vim extension.

1. Press Escape to enter Vim Command Mode
1. Type /%% to search for the beginning of a cell
1. Type N to find the top of the current cell (or g to jump to beginning of file)
1. Type Shift+V to select the current line
1. Type n to jump to beginning of next cell (or G to jump to end of file)
1. Use mapped keybinding to run command "extension.runMatlabSelection"

I have also mapped <Ctrl+A> to toggle focus between the editor and terminal.

**Comment by weihongliang233** [comment]
> Yes, similar to that. The idea of code blocks or code cells. The Matlab IDE does this itself using `%%`.
> There is similar functionality in Python IDEs (Jupyter, Spyder, VSCode Python extension) with some delimiter such as `#%%`.
> 
> One more thing I'd add on is to have the focus stay in the editor when code (selection, line, block, etc.) is run. Currently it always takes me to the terminal and I have to click back into the editor to run the next line.
> 
> These would be nice features. I'm not sure how complicated they would be.

Check out this issue/pullrequest [Adding "Cursor Back to Editor" function for "Run Current Selection"](https://github.com/apommel/vscode-matlab-interactive-terminal/issues/23). It solves your second problem.

**Comment by weihongliang233** [comment]
I think "splitting code into blocks" should be an independent functionality. I'm working on it. Here is a [demo](https://github.com/weihongliang233/code-block/blob/main/img/demo.gif). I'll integrate it into this extension soon and hope can implement the "run code blocks" functionality.

**Comment by weihongliang233** [comment]
Almost done (if there is no unexpected bugs).
